pulser/waveforms.py

```diff
@@ -429,12 +429,16 @@ def __init__(self, duration: Union[int, Parametrized],
                  area: Union[float, Parametrized]):
         """Initializes a Blackman waveform."""
         super().__init__(duration)
-        if isinstance(area, float):
-            self._area: float = area
-            self._norm_samples: np.ndarray = np.clip(
-                np.blackman(self._duration), 0, np.inf)
-            self._scaling: float = self._area / \
-                float(np.sum(self._norm_samples)) / 1e-3
+        try:
+            self._area: float = float(cast(float, area))
```

Comment:
Do you need the `cast` here?

Reply:
Nevermind, I can see you do.

Reply:
> Do you need the `cast` here?

According to MyPy, I do: `pulser\waveforms.py:433: error: Argument 1 to "float" has incompatible type "Union[float, Parametrized]"; expected "Union[SupportsFloat, _SupportsIndex, str]"`.